src/write.rs

```diff
@@ -748,4 +748,46 @@ mod tests {
         assert!(w.oplog[w.swap_index..].is_empty());
         w.flush();
     }
+
+    #[test]
+    fn flush_no_refresh() {
+        let x = 'x';
+
+        let (_, mut w) = new();
+
+        // Until we refresh, writes are written directly instead of going to the
+        // oplog (because there can't be any readers on the w_handle table).
+        w.refresh();
+
+        assert_eq!(w.second, true);
+
+        w.flush();
+
+        // No refresh because there are no operations
+        assert_eq!(w.second, true);
+
+        w.insert(x, x);
+        w.flush();
+
+        // A refresh happened!
+        assert_eq!(w.second, false);
+
+        // Reset this flag so we can see if another refresh happens
+        w.second = true;
```

Comment:
I wonder what kind of weird, deeper effects changing this externally might have... Hopefully none :sweat_smile:

Reply:
Haha yeah this feels ... horrible ... but I'm not sure how else to check, "Did a refresh happen"? Is there some other flag/meta data I could check?

Reply:
To be honest, I'd probably just add a:
```rust
#[cfg(test)]
refreshes: usize
```
field to `WriteHandle`, and then have a
```rust
#[cfg(test)]
self.refreshes += 1;
```
in the appropriate place in `refresh`. That way, in testing code, you'll know exactly how many refreshes have happened. And it wouldn't require hackery or impact running code.

Reply:
Hmmm maybe I just have the wrong syntax but I'm getting some
```
attributes on expressions are experimental
removing an expression is not supported in this position
```
when I try
```rust
pub struct WriteHandle {
    // ...
    #[cfg(test)]
    refreshes: usize,
}
pub(crate) fn new(/*...*/) -> WriteHandle {
    let m = w_handle.meta.clone();
    WriteHandle {
        // ...
        #[cfg(test)]
        refreshes: 0,
    }

  pub fn refresh(&mut self) -> &mut Self {
    // ...
    #[cfg(test)]
    self.refreshes += 1;
```
Was that what you had in mind?

Reply:
Ugh, I forgot they're still experimental. Just do this instead:
```rust
#[cfg(test)]
{
  self.refreshes += 1;
}
```
and that should work I think

Reply:
I'm a little wigged out about sprinkling the test cfg around too much but https://github.com/jonhoo/rust-evmap/pull/60 might do it